File: app/src/adapters/geo.py
# SPDX-License-Identifier: MIT
from typing import Dict, Optional

def embed_geo(frame_meta: Dict, facility_id: Optional[str] = None) -> Dict:
    geo = {
        "latitude": frame_meta.get("latitude"),
        "longitude": frame_meta.get("longitude"),
        "altitude": frame_meta.get("altitude"),
        "orbitId": frame_meta.get("orbitId"),
    }
    if facility_id:
        geo["facility_id"] = facility_id
    return geo

File: ground/ingest/src/Models/Geo.cs
// SPDX-License-Identifier: MIT
namespace ProHarp.Ingest.Models;

public class Geo
{
    // Auto telemetry (from orbital SDK)
    public double? Latitude { get; set; }
    public double? Longitude { get; set; }
    public double? Altitude { get; set; }
    public string? OrbitId { get; set; }
    public string? FacilityId { get; set; }

    // Provenance
    public string TelemetrySource { get; set; } = "auto"; // "auto" or "manual"

    // Manual override (optional)
    public ManualOverride? ManualOverride { get; set; }
}

public class ManualOverride
{
    public bool Enabled { get; set; }
    public double? Latitude { get; set; }
    public double? Longitude { get; set; }
    public string? Reason { get; set; }
}